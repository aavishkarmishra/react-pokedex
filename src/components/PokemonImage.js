import React from 'react';
import {padNumber} from '../utils';
import pokemonSymbols from '../pokemon-symbols.svg';

const fixNames = name => {
  if (name === 'sceptile') {
    return 'sceptille';
  }
  if (name === 'jirachi') {
    return 'lirachi';
  }
  if (name === 'ninjask') {
    return 'niniask';
  }
  if (name === 'baltoy') {
    return 'baltov';
  }
  if (name === 'claydol') {
    return 'clavdol';
  }
  if (name.includes('deoxys')) {
    return 'deoxys';
  }
  return name;
};

const PokemonImage = ({id, name, size = 100, animate = false}) => (
  <figure
    style={{padding: 50}}
    className={`pokemon-image image is-4by3 ${animate ? 'animate' : ''}`}
  >
    <svg style={{height: size, width: size}}>
      <use
        xlinkHref={`${pokemonSymbols}#${padNumber(id, 3)}-${fixNames(name)}`}
      />
    </svg>
  </figure>
);

export default PokemonImage;
