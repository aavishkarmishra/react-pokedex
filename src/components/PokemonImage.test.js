import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import PokemonImage from './PokemonImage';

describe('<PokemonImage />', () => {
  const defaultProps = {
    id: 1,
    name: 'bulbasaur',
  };

  const shallowComponent = customProps => {
    const props = {
      ...defaultProps,
      ...customProps,
    };
    return shallow(<PokemonImage {...props} />);
  };

  it('renders correctly', () => {
    const tree = renderer.create(<PokemonImage {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should set the correct svg link', () => {
    const wrapper = shallowComponent();
    expect(wrapper.find('use').prop('xlinkHref')).toEqual(
      '/pokemon-symbols.svg#001-bulbasaur',
    );
  });

  it('should set fix the sceptile name', () => {
    const wrapper = shallowComponent({name: 'sceptile'});
    expect(wrapper.find('use').prop('xlinkHref')).toEqual(
      '/pokemon-symbols.svg#001-sceptille',
    );
  });

  it('should set fix the jirachi name', () => {
    const wrapper = shallowComponent({name: 'jirachi'});
    expect(wrapper.find('use').prop('xlinkHref')).toEqual(
      '/pokemon-symbols.svg#001-lirachi',
    );
  });

  it('should set fix the ninjask name', () => {
    const wrapper = shallowComponent({name: 'ninjask'});
    expect(wrapper.find('use').prop('xlinkHref')).toEqual(
      '/pokemon-symbols.svg#001-niniask',
    );
  });

  it('should set fix the baltoy name', () => {
    const wrapper = shallowComponent({name: 'baltoy'});
    expect(wrapper.find('use').prop('xlinkHref')).toEqual(
      '/pokemon-symbols.svg#001-baltov',
    );
  });

  it('should set fix the claydol name', () => {
    const wrapper = shallowComponent({name: 'claydol'});
    expect(wrapper.find('use').prop('xlinkHref')).toEqual(
      '/pokemon-symbols.svg#001-clavdol',
    );
  });

  it('should set fix the deoxys name', () => {
    const wrapper = shallowComponent({name: 'deoxys-normal'});
    expect(wrapper.find('use').prop('xlinkHref')).toEqual(
      '/pokemon-symbols.svg#001-deoxys',
    );
  });
});
