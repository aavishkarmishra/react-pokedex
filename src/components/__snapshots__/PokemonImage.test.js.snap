// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonImage /> renders correctly 1`] = `
<div
  style={
    Object {
      "padding": 20,
    }
  }
>
  <img
    alt="bulbasaur"
    src="/pokemons/001-bulbasaur.svg"
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
</div>
`;
